src/app/services/product.service.ts

```diff
@@ -0,0 +1,89 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
+import { Observable, of } from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+
+import { environment } from '../../environments/environment';
+import { Product } from '../models'
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProductService {
+
+  readonly imageUrl = 'https://fruitshoppe.firebaseapp.com/images'; // URL of product images folder
+  readonly apiRoute = '/products';  // API route to retrieve products
```

Comment:
how about `productEndpoint` or something that specifies a particular path for products. This paired with my other suggestion about renaming the environment properties for API roots would ehn look something like this:
`${environment.localApiRoout}${this.productEndpoint}`

this is more readable to me.

Reply:
sure thing